src/plugins/intel_gna/gna_plugin_internal.hpp

```diff
@@ -56,8 +56,11 @@ class GNAPluginInternal  : public InferenceEngine::IInferencePlugin {
         updated_config.UpdateFromMap(config);
         auto plg = std::make_shared<GNAPlugin>(updated_config.keyConfigMap);
         plgPtr = plg;
+        auto network_impl = std::make_shared<GNAExecutableNetwork>(modelFileName, plg);
+        // set pointer for IInferencePlugin interface
+        network_impl->SetPointerToPlugin(shared_from_this());
 
-        return std::make_shared<GNAExecutableNetwork>(modelFileName, plg);
+        return network_impl;
```

Comment:
have we enabled https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/plugin/shared/include/behavior/plugin/life_time.hpp for GNA to check this behavior?

Reply:
Yes, we have those tests - https://github.com/mryzhov/openvino/blob/a79830cb55623ce2654f0685b0ca51b3f412195a/src/tests/functional/plugin/gna/shared_tests_instances/behavior/plugin/life_time.cpp#L24

Reply:
But we missed the tests for the new API, added them + added additional test covering this bug